packages/http-client/src/document.ts

```diff
@@ -78,9 +79,11 @@ class Document extends Doctype {
     return new Document(DoctypeUtils.deserializeState(state), context, apiUrl)
   }
 
-  static async load (docId: DocID | string, apiUrl: string, context: Context, config: CeramicClientConfig): Promise<Document> {
+  static async load (docId: DocID | string, apiUrl: string, context: Context, config: CeramicClientConfig, opts: DocOpts, tip?: CID): Promise<Document> {
+    // todo send opts and tip
     docId = typeDocID(docId)
     const { state } = await fetchJson(apiUrl + '/documents/' + docId.toString())
+    // todo if loading at tip, disable document syncing
```

Comment:
I don't know enough of the underlying concerns regarding the tips, bloom filter, etc. but I agree that versions shouldn't necessarily mean anchored, at least not from a consumer perspective. My basic expectation of a version is that if I make a change to a doc, it gets a new version, and I can load the contents of the doc at a certain time by specifying a version.

I also think the `_docmap` is problematic as it makes the client stateful, which I wouldn't expect for a HTTP client. Caching logic should exist at another layer, or if built-in at least be opt-in.

We've discussed this before but to me the fundamental problem is that we're mixing "frontend" (consumer) and "backend" (protocol) APIs in a single interface, so concerns from each side end-up leaking to the other.

Reply:
I think the main reasons we went with only *anchor records* being *versions* is that versions would then always be able to provide a proof of publication, i.e. when the version was created.
I'm open to explore version being a more broad concept. My main concern is that it causes issues for the `3id-did-provider` package. In there we need to be able to know which version are anchors to make sure that the key rotation is done correctly. Other than that I wonder if there might be other unforeseen consequences of a more broad definition of versions. Not sure how difficult it would be to change on the ceramic side.

Reply:
I guess the problem with **3id-did-provider** could be solved if we have a specific way of getting anchored versions.

Reply:
What's the issue with 3id-did-provider?  Would a `anchorVersionIds()` method that behaves like our current [allVersionIds()](https://github.com/ceramicnetwork/js-ceramic/blob/develop/packages/common/src/doctype.ts#L181) method be sufficient?

As for the ceramic side, I suspect it will actually simplify a lot of code in js-ceramic.  If there's no serious objections I'd like to put this together today as I think it is by far the most expedient route to unblocking the bloom filter implementation.  @oed